internal/gitops/structure/key.go

```diff
@@ -0,0 +1,76 @@
+package structure
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+const (
+	fluxKustomizationAPIVersion = "kustomize.toolkit.fluxcd.io/v1beta2"
+	fluxKustomizationKind       = "Kustomization"
+
+	directoryManagementClusters = "management-clusters"
+	directoryOrganizations      = "organizations"
+	directorySecrets            = "secrets"
+	directorySOPSPublicKeys     = ".sops.keys"
+	directoryWorkloadClusters   = "workload-clusters"
+
+	mainKustomizationSuffix     = "gitops"
+	clustersKustomizationPrefix = "clusters"
+
+	kustomizationsNamespace  = "default"
+	kustomizationsSourceKind = "GitRepository"
+)
+
+func fileName(name string) string {
+	return fmt.Sprintf("%s.yaml", name)
+}
+
+func kustomizationName(mc, wc string) string {
+	if wc == "" {
+		return fmt.Sprintf("%s-%s", mc, mainKustomizationSuffix)
+	} else {
+		return fmt.Sprintf("%s-%s-%s", mc, clustersKustomizationPrefix, wc)
+	}
+}
+
+func kustomizationPath(mc, org, wc string) string {
+	if wc == "" || org == "" {
+		return fmt.Sprintf("./%s/%s", directoryManagementClusters, mc)
+	} else {
+		return fmt.Sprintf(
+			"./%s/%s/%s/%s/%s/%s",
+			directoryManagementClusters,
+			mc,
+			directoryOrganizations,
+			org,
+			directoryWorkloadClusters,
+			wc,
+		)
+	}
+}
+
+func kustomizationManifest(name, path, repository, serviceAccount, interval, timeout string) unstructured.Unstructured {
+	return unstructured.Unstructured{
```

Comment:
Can we / makes sense to use the actual go structures for the resource descriptors we create?